### ISSUE TITLE
[Question] Pinning with SHA-2?

### ISSUE THREAD

**Issue by JackWink** [square/okhttp#1661]
I noticed a lot of the pinning code references Google's practices circa 2011 (at least, that's when Adam Langley's post is dated), but Google is [now moving away from SHA-1.](http://googleonlinesecurity.blogspot.com/2014/09/gradually-sunsetting-sha-1.html)

Is there a plan for adding SHA-2 support, i.e would that be an acceptable PR or is there a reason to only support SHA-1?

**Comment by swankjesse** [comment]
We're waiting on Chrome to support SHA-2 in its own pins before proceeding. It's currently SHA-1 only.
http://src.chromium.org/viewvc/chrome/trunk/src/net/http/transport_security_state_static.certs

**Comment by JackWink** [comment]
Appreciate the quick reply, but I guess I don't follow why that would prevent adding support for SHA-2 pinning in addition to SHA-1? I'd be happy to submit a PR if it's a matter of priorities right now... 

For what it's worth, it seems that Chrome has supported SHA-2 pinning since version 38 (https://developer.mozilla.org/en-US/docs/Web/Security/Public_Key_Pinning), it looks like they just haven't updated their own static pins.

**Comment by swankjesse** [comment]
That's HPKP. Got an example of a SHA-256 pin that's baked into the client?

**Comment by swankjesse** [comment]
(We're tracking HPKP support separately [here](https://github.com/square/okhttp/issues/1567)).

**Comment by JackWink** [comment]
Not in Chrome, but it looks like Firefox uses SHA-2 static pins... I found the patch from when they added support for importing Chrome's static pins, it looks like they use "sha256/" as their prefix. https://bug772756.bugzilla.mozilla.org/attachment.cgi?id=8421109

https://bugzilla.mozilla.org/show_bug.cgi?id=772756 has some of the conversation around adding support for both static pins.

**Comment by swankjesse** [comment]
Oh cool. In that case, I'm happy to accept code! Ideally one pull request to add support, and a second PR to change our defaults over.

**Comment by swankjesse** [comment]
Duplicate of https://github.com/square/okhttp/issues/1539 which was fixed!
